### PR TITLE
Fix diff_test to catch additions too

### PR DESCRIPTION
**Pull request by cjb**
Thanks to @mlsteele for the snippet.

r? @oconnor663 

CC @jzila

### CODE REVIEW

File: protocol/diff_test.sh [hunk 1]
```diff
@@ -16,9 +16,18 @@ make
 # This build process is idempotent. We expect there to be no changes after
 # re-running the protocol generation, because any changes should have been
 # checked in.
+
 if ! git diff --exit-code ./ ../go/ ../shared/; then
```
Comment (jzila) [inline protocol/diff_test.sh]: can `git add -A ./ ../go/ ../shared/` first, then `git diff --quiet --exit-code HEAD -- ./ ../go/ ../shared/`

Comment (jzila) [inline protocol/diff_test.sh]: Unless you want the cases to be separated, in which case the current snippet is fine

**Comment by jzila** [comment]
@cjb one more thing: [here](https://github.com/keybase/client/blob/master/Jenkinsfile#L80) we currently `git add` the revision in order to exploit the current diff_test behavior. To fix tests we'll need to figure out a cleaner way to apply that.

**Comment by oconnor663** [comment]
I think we can get away with one `git status` command to cover both of these cases.

**Comment by jzila** [comment]
LGTM. Please verify that it fails if you've changed an AVDL but not the underlying protocol.

**Comment by jzila** [comment]
FWIW I'm pretty sure that right now the test failure that's happening is because my aforementioned `revision` file created for the Docker.

**Comment by jzila** [comment]
@cjb might want to add a `git commit -am "revision file added"` [here](https://github.com/keybase/client/blob/205157ae5bcea4b3b1579037b9960627b0bd605f/Jenkinsfile#L80)

**Comment by jzila** [comment]
You can just `git commit --author="Jenkins <ci@keybase.io>" -am "revision file added"`

File: protocol/diff_test.sh [hunk 2]
```diff
@@ -9,14 +9,22 @@ npm i
...
 # Protocol changes could create diffs in the following directories:
 #   protocol/
 #   go/
 #   shared/
 # This build process is idempotent. We expect there to be no changes after
 # re-running the protocol generation, because any changes should have been
 # checked in.
-if ! git diff --exit-code ./ ../go/ ../shared/; then
+
+git diff # for testing
```
Comment (jzila) [inline protocol/diff_test.sh]: this should be `git diff HEAD --`

**Comment by cjb** [comment]
Oh, cool.  The latest one looks to be actually reporting a legit build break that was present in the version of master that I branched.  So that's our positive test.

**Comment by jzila** [comment]
LGTM

**Pull request merged**